### PR TITLE
Feat/옵션기반검색확장용공개타입추가(HangulSearchOptions, Hit, MatchKind)

### PR DESCRIPTION
**Pull request by Juhwa-Lee1023**
## Description of the Change

옵션 기반 검색 API 확장을 위한 공개 타입을 Additive 방식으로 추가했습니다.

- `HangulSearchOptions` 추가
  - `mode`, `sortMode`, `limit`, `offset`, `minInputLength`, `normalizeToNFC`, `emptyQueryBehavior` 포함
  - 기본값은 기존 1.x 동작을 유지하도록 설정
- `HangulMatchKind` 추가
  - `fullMatch`, `chosungMatch`, `autocompleteMatch`
- `HangulSearchHit<Item>` 추가
  - `item`, `matchKinds`, `matchPosition`, `editDistance`를 담는 상세 결과 타입
- 공개 타입 계약 테스트 추가 (`HangulSearchPublicTypesTests`)
  - 기본값 검증
  - 커스텀 값 저장 검증
  - `HangulSearchHit` 메타데이터 저장 검증

## Motivation

- v1.1.0 Additive 기능(`searchItems(input:options:)`, `searchHits`) 도입 전 타입 계약을 먼저 고정
- 기존 API/기본 동작을 유지한 채 확장 가능한 기초 타입을 선반영
- 이후 단계에서 발생할 회귀를 타입 단위에서 조기 차단

## 핵심 변경 코드

```swift
public struct HangulSearchOptions: Hashable {
    public enum EmptyQueryBehavior: Hashable {
        case returnEmpty
        case returnAll
    }

    public var mode: HangulSearchMode?
    public var sortMode: SortMode?
    public var limit: Int?
    public var offset: Int
    public var minInputLength: Int
    public var normalizeToNFC: Bool
    public var emptyQueryBehavior: EmptyQueryBehavior
}
```

```swift
public struct HangulSearchHit<Item> {
    public let item: Item
    public let matchKinds: Set<HangulMatchKind>
    public let matchPosition: Int?
    public let editDistance: Int?
}
```

## Test

- `swift test --parallel`
- 결과: `22 tests, 0 failures`

### CODE REVIEW

File: Tests/HangulSearchTests/HangulSearchPublicTypesTests.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import XCTest
+@testable import HangulSearch
```
[the remark this answers is not in the document]
Comment (Juhwa-Lee1023) [reply Tests/HangulSearchTests/HangulSearchPublicTypesTests.swift]: 리뷰 감사합니다. 지적해주신 내용이 맞아 public API 계약 테스트에서는 `@testable import`가 부적절하다고 판단해 `import HangulSearch`로 변경했습니다.

검증:
- `swift test --parallel`
- `swift test -c release`
모두 통과했습니다.

반영 커밋: `d538eda`

File: Sources/HangulSearch/Source/Model/HangulSearchOptions.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        emptyQueryBehavior: EmptyQueryBehavior = .returnEmpty
+    ) {
+        self.mode = mode
+        self.sortMode = sortMode
+        self.limit = limit
+        self.offset = offset
+        self.minInputLength = minInputLength
+        self.normalizeToNFC = normalizeToNFC
+        self.emptyQueryBehavior = emptyQueryBehavior
+    }
```
[the remark this answers is not in the document]
Comment (Juhwa-Lee1023) [reply Sources/HangulSearch/Source/Model/HangulSearchOptions.swift:33]: 좋은 지적 감사합니다. 다만 이번 PR은 **공개 타입 추가 및 계약 고정** 범위로 제한하고 있어, initializer 단계의 값 검증/실패 정책(precondition)은 이번 변경에 포함하지 않았습니다.

`offset/limit/minInputLength`의 경계값 처리 정책은 옵션이 실제로 적용되는 `searchItems(input:options:)` 구현(PR-2)에서 동작 단위로 확정하고 테스트와 함께 반영하겠습니다.

File: Sources/HangulSearch/Source/Model/HangulSearchHit.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    public init(
+        item: Item,
+        matchKinds: Set<HangulMatchKind>,
+        matchPosition: Int? = nil,
+        editDistance: Int? = nil
+    ) {
+        self.item = item
+        self.matchKinds = matchKinds
+        self.matchPosition = matchPosition
+        self.editDistance = editDistance
```
[the remark this answers is not in the document]
Comment (Juhwa-Lee1023) [reply Sources/HangulSearch/Source/Model/HangulSearchHit.swift:18]: 리뷰 감사합니다. 이 부분도 타당한 고민입니다.

다만 현재 PR은 타입 선언 단계라 `HangulSearchHit` 자체에 precondition을 넣기보다는, 실제 생산 지점(`searchHits`)에서 `matchKinds`가 비어있지 않도록 보장하는 방향이 더 안전하다고 판단했습니다. 해당 검증은 `searchHits` 구현 PR에서 테스트와 함께 반영하겠습니다.

File: Tests/HangulSearchTests/HangulSearchPublicTypesTests.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            matchPosition: 1,
+            editDistance: 2
+        )
+        
+        XCTAssertEqual(hit.item.name, "홍길동")
+        XCTAssertEqual(hit.item.age, 30)
+        XCTAssertEqual(hit.matchKinds, [.fullMatch, .autocompleteMatch])
+        XCTAssertEqual(hit.matchPosition, 1)
+        XCTAssertEqual(hit.editDistance, 2)
+    }
```
[the remark this answers is not in the document]
Comment (Juhwa-Lee1023) [reply Tests/HangulSearchTests/HangulSearchPublicTypesTests.swift:51]: 제안 감사합니다. 현재 PR에서는 타입 계약(기본값/필드 보존)만 고정하고, invalid 입력 처리 정책은 아직 확정하지 않아 해당 edge case 테스트를 포함하지 않았습니다.

해당 경계값 테스트는 옵션 실제 적용 로직(PR-2)에서 정책 확정 후 추가하겠습니다.
